lib/services/sign_up_service.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:dio/dio.dart';
+import 'package:flutter_modular/flutter_modular.dart';
+
+class SignUpService {
+  Future<void> signUp(String name, String email, String password, String passwordConfirmation) async {
+    final api = Modular.get<Dio>();
+    await api.post('/api/signup', data: {
```

Comment:
No need to include the prefix /api. This is already being added by your env var API_BASE_URL